src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -227,6 +227,8 @@ private static ReportDiagnostic MapSeverityToReport(DiagnosticSeverity severity)
                     return ReportDiagnostic.Warn;
                 case DiagnosticSeverity.Error:
                     return ReportDiagnostic.Error;
+                case DiagnosticSeverity.Suggestion:
```

Comment:
Because this is actually a new severity that falls between warning and info. Note that "unnecessary" is orthogonal to severity. It's completely valid to have a diagnostic that is faded as unnecessary yet still squiggled as a warning or error.